File: shopy-api/Models/PaymentMethod.cs
namespace shopy_api.Models;

/// <summary>
/// Metode pembayaran yang benar-benar didukung lewat integrasi Midtrans Core API
/// di app ini. Mockup (07_pembayaran_pilih_metode.png) juga menampilkan Transfer
/// Mandiri, OVO, dan DANA — sengaja tidak diimplementasikan karena butuh
/// endpoint/alur Midtrans yang berbeda (Mandiri pakai "echannel" bukan
/// "bank_transfer", OVO butuh linking nomor HP, DANA tidak tersedia langsung
/// di Core API) dan di luar scope saat ini.
/// </summary>
public enum PaymentMethod
{
    BcaVa,
    BniVa,
    GoPay,
    Qris,
}

File: shopy-mobile/lib/models/payment/payment_method.dart
/// Metode pembayaran yang didukung, sesuai enum `PaymentMethod` di backend.
///
/// Mockup (`07_pembayaran_pilih_metode.png`) juga menampilkan Transfer Mandiri,
/// OVO, dan DANA — sengaja tidak diimplementasikan karena Midtrans Core API
/// butuh endpoint/alur berbeda untuk itu (lihat catatan di
/// `shopy-api/Models/PaymentMethod.cs`).
enum PaymentMethod {
  bcaVa('BcaVa', 'Transfer Bank BCA', 'BCA', 'Transfer Bank'),
  bniVa('BniVa', 'Transfer Bank BNI', 'BNI', 'Transfer Bank'),
  goPay('GoPay', 'GoPay', 'GP', 'E-Wallet'),
  qris('Qris', 'QRIS (semua e-wallet & bank)', 'QR', 'QRIS');

  final String apiValue;
  final String label;
  final String badge;
  final String group;

  const PaymentMethod(this.apiValue, this.label, this.badge, this.group);
}
